lambda/matching.ml

```diff
@@ -3100,6 +3257,50 @@ let rec compile_match repr partial ctx (m : pattern_matching) =
       (bind_check str v arg lam, total)
   | _ -> assert false
 
+and compile_simplified repr partial ctx (m : Simple.clause pattern_matching) =
+  match m with
+  | { cases = []; args = [] } -> comp_exit ctx m
+  | { args = ((Lvar v as arg), str) :: argl } ->
```

Comment:
The local precondition `args = ((Lvar v as arg), str) :: argl ` is new. But `compile_simplified` is only called in `do_compile_matching` which is only called by `compile_match` and `compile_half_compiled` that respect this precondition.